optuna/study/_tell.py

```diff
@@ -172,6 +172,9 @@ def _tell_with_warning(
         len(study.directions), values, trial_number
     )
 
+    if state == TrialState.COMPLETE and values_conversion_failure_message is not None:
+        raise ValueError(values_conversion_failure_message)
+
     if state is None:
```

Comment:
Not a strong opinion, but how about connecting this if statement with the next one?
The latter one also checks the `state` and `value_conversion_failure_message`. This function has a bit complicated value checking, and I'd like to reorganize the structure if possible.
```suggestion
    if state == TrialState.COMPLETE:
        if values_conversion_failure_message is not None:
            raise ValueError(values_conversion_failure_message)
    elif state is None:
```

Reply:
> This function has a bit complicated value checking, and I'd like to reorganize the structure if possible.

I agree, but may I work with another PR? I have a refactoring plan. It can be complicated to fix and refactor in same PR.

Reply:
That makes sense. Thank you.